nilearn/input_data/nifti_labels_masker.py

```diff
@@ -205,51 +251,34 @@ def transform(self, imgs, confounds=None):
         """
         self._check_fitted()
 
-        logger.log("loading images: %s" %
-                   _utils._repr_niimgs(imgs)[:200], verbose=self.verbose)
-        imgs = _utils.check_niimg_4d(imgs)
+        # We handle the resampling of labels separately because the affine of
+        # the labels image should not impact the extraction of the signal.
 
+        if not hasattr(self, '_resampled_labels_img_'):
+            self._resampled_labels_img_ = self.labels_img_
         if self.resampling_target == "data":
-            if not hasattr(self, '_resampled_labels_img_'):
-                self._resampled_labels_img_ = self.labels_img_
-            if not _check_same_fov(imgs, self._resampled_labels_img_):
-                logger.log("resampling labels", verbose=self.verbose)
-                self._resampled_labels_img_ = self._cache(image.resample_img,
-                    func_memory_level=1)(
+            imgs_ = _utils.check_niimg_4d(imgs)
+            if not _check_same_fov(imgs_, self._resampled_labels_img_):
+                if self.verbose > 0:
+                    print("Resampling labels")
+                self._resampled_labels_img_ = self._cache(
+                    image.resample_img, 2)(
                         self.labels_img_, interpolation="nearest",
                         target_shape=imgs.shape[:3],
-                        target_affine=imgs.get_affine(),
-                    )
-        elif self.resampling_target == "labels":
-            self._resampled_labels_img_ = self.labels_img_
-            logger.log("resampling images", verbose=self.verbose)
-            imgs = self._cache(image.resample_img, func_memory_level=1)(
-                imgs, interpolation="continuous",
-                target_shape=self.labels_img_.shape,
-                target_affine=self.labels_img_.get_affine())
-        else:
-            self._resampled_labels_img_ = self.labels_img_
+                        target_affine=imgs.get_affine())
+
+        region_signals = self._cache(_extract_signals, 1, ignore=['verbose'])(
```

Comment:
Don't you need to add more ignored parameters, for example memory and memory_level?